### PR TITLE
Adding aggregated logs for training run

### PR DESCRIPTION
**Pull request by MayankChaturvedi**
## What this does
This change adds four aggregated profiling metrics in training logs- average policy update time, max policy update time, average data loading time, and max data loading time. 
The comments of [issue](https://github.com/huggingface/lerobot/issues/374) describe the thought process behind the metrics.

Examples:
|                | Example           |
|----------------------|-----------------|
| New log       | smpl:2K ep:3 epch:0.06 loss:3.706 grdn:94.749 lr:1.0e-05 pu_mx_av:1.472\|1.159 dl_mx_av:0.022\|0.010      |
| Old log   | step:200 smpl:2K ep:3 epch:0.06 Loss:2.806 grdn: 14.267 lr: 10e-05 updt_s:1.278 data_s:0.010  |

## How it was tested
Ran the training script
```
python lerobot/scripts/train.py     policy=act     env=aloha     env.task=AlohaInsertion-v0     dataset_repo_id=lerobot/aloha_sim_insertion_human device=cpu
```

## How to checkout & try? (for the reviewer)
Reviewers can run the above command to validate the output

### CODE REVIEW

**Review by alexander-soare — CHANGES_REQUESTED** [review]
Looking mostly good to me! Please take a review at my comments, and once done I'll try it out on my local machine.

File: lerobot/scripts/train.py [hunk 1]
```diff
@@ -175,19 +177,22 @@ def log_train_info(logger: Logger, info, step, cfg, dataset, is_online):
     num_episodes = num_samples / avg_samples_per_ep
     num_epochs = num_samples / dataset.num_samples
     log_items = [
-        f"step:{format_big_number(step)}",
         # number of samples seen during training
         f"smpl:{format_big_number(num_samples)}",
         # number of episodes seen during training
         f"ep:{format_big_number(num_episodes)}",
         # number of time all unique samples are seen
```
[the remark this answers is not in the document]
Comment (MayankChaturvedi) [reply lerobot/scripts/train.py:184]: Thanks! this helps

File: lerobot/scripts/train.py [hunk 2]
```diff
@@ -175,19 +177,22 @@ def log_train_info(logger: Logger, info, step, cfg, dataset, is_online):
...
         f"grdn:{grad_norm:.3f}",
+        # learning rate at the end of the past step
         f"lr:{lr:0.1e}",
-        # in seconds
-        f"updt_s:{update_s:.3f}",
-        f"data_s:{dataloading_s:.3f}",  # if not ~0, you are bottlenecked by cpu or io
+        # aggregated policy update time(s) in format max_update_time|average_update_time since the last log
+        f"pu_mx_av:{max_policy_updating_s:.3f}|{avg_policy_updating_s:.3f}",
+        # data loading time(s) in format max_loading_time|average_loading_time since the last log
+        f"dl_mx_av:{max_data_loading_s:.3f}|{avg_data_loading_s:.3f}",  # if not ~0, you are bottlenecked by cpu or io
```
[the remark this answers is not in the document]
Comment (MayankChaturvedi) [reply lerobot/scripts/train.py:195]: Great suggestion. I'll also add milliseconds idea to data_max|avg

**Comment by alexander-soare** [comment]
Thanks @MayankChaturvedi 

I tried this out, and I think maybe the pipe symbol is quite jarring on the eyes trying to scan the line quickly. Maybe a `,` looks better?

```
# Current version.
INFO 2024-09-06 10:28:34 ts/train.py:197 smpl:3K ep:26 epch:0.13 loss:0.391 grdn:8.145 lr:1.0e-05 updt_max|avg:89|84 data_max|avg:106|28

# Suggestion.
# Here I also add "ms" to make it clear it's milliseconds. To make space for that, I removed the "ep" entry.
INFO 2024-09-06 10:28:34 ts/train.py:197 smpl:3K epch:0.13 loss:0.391 grdn:8.145 lr:1.0e-05 updt_ms_max,avg:89,84 data_ms_max,avg:106,28
```

Let's also get @Cadene's input

**Comment by Cadene** [comment]
Hello @MayankChaturvedi ,
Thanks a lot of your contribution on this PR.
We will come back to it when time allows.
Thanks for your understanding.
Best
